packages/api-v4/src/cloudpulse/types.ts

```diff
@@ -116,3 +116,89 @@ export interface CloudPulseMetricsList {
   metric: { [resourceName: string]: string };
   values: [number, string][];
 }
+
+export interface CreateAlertDefinitionPayload {
+  name: string | null;
+  region: string | null;
+  description?: string | null;
+  service_type: string | null;
+  engineOption: string | null;
+  resource_ids: string[];
+  severity: string;
+  criteria: MetricCriteria[];
+  triggerCondition: {
+    criteria_condition: string;
+    polling_interval_seconds: string;
+    evaluation_period_seconds: string;
+    trigger_occurrences: number;
+  };
+  notifications: {
+    notification_type: string;
+    content: {
+      email: string;
+    };
+  }[];
+  sink_ids: string[];
+}
+export interface MetricCriteria {
+  metric: string;
+  aggregation_type: string;
+  operator: string;
+  value: number;
+  dimension_filters: {
+    dimension_label: string;
+    operator: string;
+    value: string;
+  }[];
+}
+
+export interface Alert {
+  id: number;
+  name: string;
+  description: string;
+  status: string;
+  severity: string;
+  service_type: string;
+  resource_ids: string[];
+  criteria: MetricCriteria[];
+  triggerCondition: {
+    criteria_condition: string;
+    polling_interval_seconds: string;
+    evaluation_period_seconds: string;
+    trigger_occurrences: number;
+  };
```

Comment:
```suggestion
  triggerCondition: TriggerCondition;
```

use the interfaces created above and also convert other bigger nested objects to a interface